server/service/http_realip.go

```diff
@@ -0,0 +1,30 @@
+package service
+
+import (
+	"net/http"
+	"strings"
+)
+
+// copied from https://github.com/go-chi/chi/blob/c97bc988430d623a14f50b7019fb40529036a35a/middleware/realip.go#L42
+
+var trueClientIP = http.CanonicalHeaderKey("True-Client-IP")
+var xForwardedFor = http.CanonicalHeaderKey("X-Forwarded-For")
+var xRealIP = http.CanonicalHeaderKey("X-Real-IP")
+
+func realIP(r *http.Request) string {
+	var ip string
+
+	if tcip := r.Header.Get(trueClientIP); tcip != "" {
+		ip = tcip
+	} else if xrip := r.Header.Get(xRealIP); xrip != "" {
+		ip = xrip
+	} else if xff := r.Header.Get(xForwardedFor); xff != "" {
+		i := strings.Index(xff, ",")
+		if i == -1 {
+			i = len(xff)
+		}
+		ip = xff[:i]
+	}
+
```

Comment:
I suppose this means the instance of our own http server?

```	
        // RemoteAddr allows HTTP servers and other software to record
	// the network address that sent the request, usually for
	// logging. This field is not filled in by ReadRequest and
	// has no defined format. The HTTP server in this package
	// sets RemoteAddr to an "IP:port" address before invoking a
	// handler.
	// This field is ignored by the HTTP client.
	RemoteAddr string
```

given by: https://pkg.go.dev/net/http

From the examples I have seen, most middleware tends to use the approach of extracting directly from headers.

Reply:
I am definitely no expert, but I saw that Chi uses that field if they don't parse successfully from a header: https://github.com/go-chi/chi/blob/c97bc988430d623a14f50b7019fb40529036a35a/middleware/realip.go#L34

I am finding that documentation a bit vague, but does this suggest it would be set?
> The HTTP server in this package sets RemoteAddr to an "IP:port" address before invoking a handler.

Reply:
oh I missed that, great catch, will update!

Reply:
actually it seems as though they are populating `r.RemoteAddr` if they find a non empty value via `realIP(r)`

Reply:
Doh! I completely misread.